ratis-server/src/main/java/org/apache/ratis/server/impl/SnapshotInstallationHandler.java

```diff
@@ -172,6 +174,12 @@ private InstallSnapshotReplyProto checkAndInstallSnapshot(InstallSnapshotRequest
       state.setLeader(leaderId, "installSnapshot");
 
       server.updateLastRpcTime(FollowerState.UpdateType.INSTALL_SNAPSHOT_START);
+      if (snapshotChunkRequest.getRequestIndex() == 0) {
+        chunkRequestIndex.set(0);
```

Comment:
Should it be `-1`?  It should be incremented only after `state.installSnapshot(..)` has succeeded.

Reply:
After successfully receiving the first chunk, the chunkRequestIndex is 0,  but the requestIndex of the second chunk will be 1.

Reply:
@133tosakarin , I see.  The if-condition is also changed.  You are right that both cases are correct.